### ISSUE TITLE
'children' is missing in props validation(react/prop-types) when using `React.PropsWithChildren` as of 7.25.0

### ISSUE THREAD

**Issue by FritsvanCampen** [yannickcr/eslint-plugin-react#3062]
When you have a typescript component like:

```
const MyComponent = (props: React.PropsWithChildren<{}>): React.ReactElement => {
	return <>{props.children}</>;
};
```

As of 7.25 you get the warning:

> 'children' is missing in props validation(react/prop-types)

`React.PropsWithChildren` is defined as: 
```
type PropsWithChildren<P> = P & { children?: ReactNode | undefined };
```

If you change the Component props type to:

```
const MyComponent = (props: React.PropsWithChildren<{ children?: ReactNode | undefined }>): React.ReactElement => {
	return <>{props.children}</>;
};
```

The warning disappears.

I believe this is a mistake. The warning should not trigger in both cases.
It did not trigger in 7.24. It does as of 7.25.

**Comment by ljharb** [comment]
cc @Wesitos

**Comment by vedadeepta** [comment]
Not sure if this is a bug, `children` is indeed not defined right? @ljharb @FritsvanCampen

**Comment by vedadeepta** [comment]
Will look into it

**Comment by johnhunter** [comment]
I'm seeing the same with `React.FC` when upgrading from 7.24.0 to 7.25.0

**Comment by vedadeepta** [comment]
@johnhunter the PR fixes `React.FC` also
this should work
```
          import React from 'react';
          const Person: React.FC<{}> = (props): React.ReactElement => (
              <>{props.children}</>
          );

```

**Comment by ljharb** [comment]
Fixed in #3064, which will be released soon.

**Comment by FritsvanCampen** [comment]
Thanks for the quick fix!

**Comment by johnhunter** [comment]
I still see `'children' is missing in props validation  react/prop-types` using `React.FC` with v7.25.1

**Comment by ljharb** [comment]
@johnhunter you’ll see that for eternity, since that version doesn’t contain the fix.
